tutorials/src/bin/basic-tutorial-5.rs

```diff
@@ -0,0 +1,377 @@
+            .get_buffer()
+            .expect("Couldn't get buffer from text_view"),
+    );
+    playbin.get_bus().unwrap().connect_message(move |_, msg| {
+        //let streams_view = &streams_view;
+        match msg.view() {
+            gst::MessageView::Application(_) => if msg.get_structure().get_name() == "tags-changed"
+            {
+                analyze_streams(&pipeline, &textbuf);
+            },
+            _ => (),
+        }
+    });
+
+    let vbox = Box::new(Orientation::Horizontal, 0);
+    vbox.pack_start(&video_window, true, true, 0);
+    vbox.pack_start(&streams_list, false, false, 2);
+
+    let main_box = Box::new(Orientation::Vertical, 0);
+    main_box.pack_start(&vbox, true, true, 0);
+    main_box.pack_start(&controls, false, false, 0);
+    main_window.add(&main_box);
+    main_window.set_default_size(640, 480);
+
+    main_window.show_all();
+}
+
+#[cfg(feature = "tutorial5")]
+// We are possibly in a GStreamer working thread, so we notify the main
+// thread of this event through a message in the bus
+fn post_app_message(playbin: &gst::Element) {
+    let mbuilder = gst::Message::new_application(gst::Structure::new_empty("tags-changed"));
+    playbin.post_message(&mbuilder.build());
+}
+
+#[cfg(feature = "tutorial5")]
+fn main() {
+    // Make sure the right features were activated
+    if !cfg!(feature = "tutorial5-x11") && !cfg!(feature = "tutorial5-quartz") {
+        eprintln!("No Gdk backend selected, compile with --features tutorial5[-x11][-quartz].");
+
+        return;
+    }
+
+    // Initialize GTK
+    if gtk::init().is_err() {
+        eprintln!("Failed to initialize GTK.");
+        return;
+    }
+
+    // Initialize GStreamer
+    if gst::init().is_err() {
+        eprintln!("Failed to initialize GTK.");
+        return;
+    }
+
+    let uri = "https://www.freedesktop.org/software/gstreamer-sdk/\
+               data/media/sintel_trailer-480p.webm";
+    let playbin = gst::ElementFactory::make("playbin", None).unwrap();
+    if playbin
+        .set_property("uri", &glib::Value::from(uri))
+        .is_err()
+    {
+        eprintln!("Could not set URL");
+        return;
+    }
+
+    let pipeline = playbin.clone();
+    playbin
+        .connect("video-tags-changed", false, move |_| {
+            post_app_message(&pipeline);
+            None
+        })
+        .unwrap();
+
+    let pipeline = playbin.clone();
+    playbin
+        .connect("audio-tags-changed", false, move |_| {
+            post_app_message(&pipeline);
+            None
+        })
+        .unwrap();
+
+    let pipeline = playbin.clone();
+    playbin
+        .connect("text-tags-changed", false, move |_| {
+            post_app_message(&pipeline);
+            None
+        })
+        .unwrap();
+
+    create_ui(&playbin);
+    playbin.set_state(gst::State::Playing);
+
+    let bus = playbin.get_bus().unwrap();
+    let pipeline = playbin.clone();
+    bus.add_watch(gst::Bus::async_signal_func);
```

Comment:
Isn't that the same as `bus.add_signal_watch()`?

Reply:
lol, not sure how I forgot and ended up adding the API :-)

Removed Bus::async_signal_func it should not be so useful, let me know if you want me to open a PR to re add it.

Reply:
No that's fine :) I didn't add it before because it seemed a bit useless